### ISSUE TITLE
How to hide build warnings for local development

### ISSUE THREAD

**Issue by jakeleventhal** [logtail/logtail-nextjs#6]
```
warn - logtail: Envvars not detected. If this is production please see https://github.com/logtailhq/next-logtail for help
warn - logtail: Sending Web Vitals to /dev/null
warn - logtail: Sending logs to console
```

whenever i build my app locally, i get the following errors. How do i suppress these?

**Comment by curusarn** [comment]
Hi @jakeleventhal,

Thank you for raising this!

These warnings are supposed to help people avoid forgetting to specify important ENV vars in prod and losing logs. We want to be transparent to prevent mistakes. We could add another ENV variable to suppress these warnings.

Are the warnings annoying enough during development that you want to disable them?

Thanks again for reaching out.

**Comment by jakeleventhal** [comment]
Yes @curusarn, i think maybe if node_env is not production they should be suppressed

**Comment by alexmarmon** [comment]
@curusarn I'm with @jakeleventhal on this and would love a way to suppress the warnings

**Comment by curusarn** [comment]
Thank you for the feedback!

We're tracking the issue internally. We should get to fixing it soon.
I'll make sure to keep you updated here!

**Comment by PetrHeinz** [comment]
Hello @jakeleventhal and @alexmarmon, thanks for pushing this! 🙌 

Just released [`v0.1.2`](https://github.com/logtail/logtail-nextjs/releases/tag/v0.1.2) that solves the issue by suppressing the warning about missing ENV vars in development environment. You can even override this behavior via `LOGTAIL_CHECK_ENV_VARS ` env var. For details, see #12

Hope that helps! 🙂

**Comment by jakeleventhal** [comment]
On my local environment with `0.1.2` i get this output with no env vars detected:
```
warn - logtail: Envvars not detected. If this is production please see https://github.com/logtail/logtail-nextjs for help
warn - logtail: Sending Web Vitals to /dev/null
warn - logtail: Sending logs to console
```

**Comment by PetrHeinz** [comment]
Seems you have set different `NODE_ENV` than `development`.

Could you try running it with `LOGTAIL_CHECK_ENV_VARS =false`?

**Comment by jakeleventhal** [comment]
Yeah it works. when i run build it just default to NODE_ENV=production?

all i am doing is running `pnpm run build` which just runs the script for `vercel build`

I think this is default vercel behavior. maybe `process.env.NODE_ENV` isn't the best proxy for this if that is the default

**Comment by PetrHeinz** [comment]
It worked nicely with my setup (without involving Vercel). Now at least, the detection can be suppressed for local development, which is a good thing 🙂

**Comment by jakeleventhal** [comment]
Yeah, but I think a better solution ought to exist considering vercel tooling is super common with next.js
I tried setting `LOGTAIL_CHECK_ENV_VARS` to `false` in my vercel config but it treats `false` as `"false"` and therefore doesnt work

**Comment by PetrHeinz** [comment]
`"false"` is actually OK. ENV vars always contain strings. The `LOGTAIL_CHECK_ENV_VARS ` should work with `1`, `0`, `true` and `false` as most common representation of booleans. See #12 for more details.

If it doesn't work for you, the ENV vars are probably not accessible in your app.

**Comment by jakeleventhal** [comment]
the env var is accessible in my app it just doesnt seem to be accessible in `vercel build`

what do you propose? it seems overkill to ask all vercel users to maintain a .env file specifically for suppressing logtail logs

**Comment by PetrHeinz** [comment]
Hi @jakeleventhal, thanks for the follow-up.

You're right, in Vercel there's different way of determining the app environment (via [`VERCEL_ENV`](https://vercel.com/docs/projects/environment-variables/system-environment-variables)).

I'm addressing this in #14, after release of the new version, you shouldn't see the warning during `vercel build` ran in development 👍

**Comment by PetrHeinz** [comment]
@jakeleventhal Interesting... 🤔 Thanks for the further details on this. What's your `vercel --version`?

I tested the change with Vercel CLI 32.5.6

**Comment by PetrHeinz** [comment]
Sorry, I can see that in your log output - Vercel CLI 32.5.3

**Comment by PetrHeinz** [comment]
@jakeleventhal I'm having difficulty replicating it. Could you please check that https://github.com/logtail/logtail-nextjs/pull/15 works in your case?

**Comment by jakeleventhal** [comment]
Is there a canary release or a compiled dist file? Having trouble generating it locally

**Comment by PetrHeinz** [comment]
@jakeleventhal There's no pre-release at the moment, I assumed you could edit the dist manually, just as you did in https://github.com/logtail/logtail-nextjs/issues/6#issuecomment-1824597088.

**Comment by jakeleventhal** [comment]
This one is a bit trickier as the compiled js is quite different and ther are a lot of conditions here. can you provide a snippet of the compiled `warnAboutMissingEnvironmentVariables` function

**Comment by PetrHeinz** [comment]
From `dist/withLogtail.js`:
 
 ```js
 function warnAboutMissingEnvironmentVariables() {
    var _a, _b;
    var nodeDevEnv = process.env.NODE_ENV === 'development';
    var vercelDevEnv = process.env.VERCEL_ENV === 'preview' || process.env.VERCEL_ENV === 'development';
    var vercelNonProdBuild = process.env.VERCEL_ENV !== 'production' && process.env.__VERCEL_BUILD_RUNNING === '1';
    var checkEnabled = !nodeDevEnv && !vercelDevEnv && !vercelNonProdBuild;
    if (((_a = process.env.LOGTAIL_CHECK_ENV_VARS) === null || _a === void 0 ? void 0 : _a.toLowerCase()) === 'true' || process.env.LOGTAIL_CHECK_ENV_VARS === '1') {
        checkEnabled = true;
    }
    if (((_b = process.env.LOGTAIL_CHECK_ENV_VARS) === null || _b === void 0 ? void 0 : _b.toLowerCase()) === 'false' || process.env.LOGTAIL_CHECK_ENV_VARS === '0') {
        checkEnabled = false;
    }
    if (checkEnabled) {
        var log = new logger_1.Logger();
        log.warn('logtail: Envvars not detected. If this is production please see https://github.com/logtail/logtail-nextjs for help');
        log.warn('logtail: Sending Web Vitals to /dev/null');
        log.warn('logtail: Sending logs to console');
    }
}
```

**Comment by jakeleventhal** [comment]
works for me, good to merge that PR

**Comment by PetrHeinz** [comment]
Thanks for the cooperation on this one, @jakeleventhal 👍 Just released [v0.1.4](https://github.com/logtail/logtail-nextjs/releases/tag/v0.1.4) with the verified fix 🚀

**Comment by jakeleventhal** [comment]
I just verified `0.1.4` gives me no issues
